src/tests/ftest/launch.py

```diff
@@ -2500,7 +2462,8 @@ def _archive_files(self, summary, hosts, source, pattern, destination, depth, th
 
         if test and "core files" in summary:
             # Process the core files
-            return_code |= self._process_core_files(os.path.split(destination)[0], test)
+            return_code |= self._process_core_files(os.path.split(destination)[0] + os.path.sep,
+                                                    test)
```

Comment:
It's not so much a case of _needed_ but more just a case about being explicit that a target is a directory.  It prevents cases such as successfully doing a `cp foo bar` where `bar/` doesn't exist yet and you carete a file instead of [failing] copying into a [missing] directory.